bindings/generate_bind_c11_extras.c

```diff
@@ -90,13 +90,64 @@
            unsigned long long: 0, \
            default: 1)
 
+/* Note: Mirrors SHMEM_BIND_C11_COLL_AND_OR_XOR */
+#define NEED_BITWISE_REDUCE_ASSOC(VAL) \
+  _Generic((VAL),                      \
+           unsigned char: 0,           \
+           unsigned short: 0,          \
+           unsigned int: 0,            \
+           unsigned long: 0,           \
+           unsigned long long: 0,      \
+           default: 1)
+
+/* Note: Mirrors SHMEM_BIND_C11_COLL_MIN_MAX */
+#define NEED_MIN_MAX_REDUCE_ASSOC(VAL) \
+  _Generic((VAL),                      \
+           char: 0,                    \
+           signed char: 0,             \
+           short: 0,                   \
+           int: 0,                     \
+           long: 0,                    \
+           long long: 0,               \
+           unsigned char: 0,           \
+           unsigned short: 0,          \
+           unsigned int: 0,            \
+           unsigned long: 0,           \
+           unsigned long long: 0,      \
+           float: 0,                   \
+           double: 0,                  \
+           long double: 0,             \
+           default: 1)
+
+/* Note: Mirrors SHMEM_BIND_C11_COLL_SUM_PROD */
+#define NEED_SUM_PROD_REDUCE_ASSOC(VAL) \
+  _Generic((VAL),                       \
+           char: 0,                     \
+           signed char: 0,              \
+           short: 0,                    \
+           int: 0,                      \
+           long: 0,                     \
+           long long: 0,                \
+           unsigned char: 0,            \
+           unsigned short: 0,           \
+           unsigned int: 0,             \
+           unsigned long: 0,            \
+           unsigned long long: 0,       \
+           float: 0,                   \
+           double: 0,                  \
+           long double: 0,             \
+           default: 1)
```

Comment:
@davidozog, Functionality wise it makes sense, but should we keep just one of them?